### PR TITLE
Fix: make log panel more asynchronous

### LINKED ISSUE #901: live_panel_output sometimes is out of sync
the log panel is still buggy

When I do `review pull request`, the output is out of sync

```
> /usr/local/bin/git fetch https://github.com/ydsdsa/AlignTab.git patch-1

[Done in 0.36s]From https://github.com/ydsdsa/AlignTab
 * branch            patch-1    -> FETCH_HEAD

```

Relevant settings
```
    "show_panel_for": [
        "commit",
        "apply",
        "fetch",
        "pull",
        "push"
    ],
    "live_panel_output": true
```

I have concluded that it is due to a synchronization issue between the io threads and the process thread. https://github.com/divmain/GitSavvy/blob/797f79ba9aa64c94efe2722eae49344b6655a66c/core/git_command.py#L106-L107

I don't have more time to further investigate now.

### PR DESCRIPTION
**Pull request by randy3k**
fix #901

### CODE REVIEW

File: core/git_command.py [hunk 1]
```diff
@@ -65,15 +65,15 @@ def __init__(self, process):
 
     def read_stdout(self):
         try:
-            for line in self.process.stdout:
+            for line in iter(self.process.stdout.readline, b""):
```
Comment (stoivo) [inline core/git_command.py]: Can you explain what you do here?

Comment (randy3k) [reply core/git_command.py]: The original code prints the buffer by bytes and I changed it to print the buffer by lines as what it is supposed to do

Comment (stoivo) [reply core/git_command.py]: ok, What are the arguments to iter?

Comment (randy3k) [reply core/git_command.py]: https://docs.python.org/3/library/functions.html#iter

Comment (stoivo) [reply core/git_command.py]: So this will read until it gets an empty line?

Comment (randy3k) [reply core/git_command.py]: No. It will read until the end of the buffer. `readline` will return `b"\n"` for an empty line.

Comment (stoivo) [reply core/git_command.py]: ok,

**Pull request merged**